dali/python/nvidia/dali/pipeline.py

```diff
@@ -297,6 +296,15 @@ def _release_outputs(self):
             raise RuntimeError("Pipeline must be built first.")
         return self._pipe.ReleaseOutputs()
 
+    def _outputs(self):
+        """Release buffers previously returned and returns  the calls.
+
+        Calling this function is equivalent to calling _release_outputs
+        then calling _share_outputs"""
+        if not self._built:
+            raise RuntimeError("Pipeline must be built first.")
+        return self._pipe.Outputs()
```

Comment:
Outputs for the pipeline does _release_outputs() and then _share_outputs(). I don't see any need to create `_outputs(self)` in python for that.

Reply:
But in executor implem  Outputs in basically calling _release_outputs and _share_outputs consecutively. Which reduces the number of Python calls to do the same thing.

https://github.com/NVIDIA/DALI/blob/6480f3bd3de9bf2653c456046613c1d4af1f64f1/dali/pipeline/executor/executor.h#L462

Reply:
Does it make such difference?

Reply:
Not really from a performance standpoint, but conceptually I think that directly using the atomic version is better here

Reply:
And also for readability, it made me wonder if this separation into 2 functions had some meaning here

Reply:
Agree